app/src/components/image/image.jsx

```diff
@@ -0,0 +1,194 @@
+import { useEffect, useState } from "react"
+import styles from "./image.module.css"
+
+import Cookies from "universal-cookie"
+
+async function getImageUrl (id) {
+    const url = `${import.meta.env.VITE_BACKEND_URL}/generate/${id}`
+
+    let options = {
+        method: "GET",
+        headers: {
+            'accept': 'application/json',
+            "Content-Type": "application/json",
+        }
+    }
+
+    const response = await fetch(url, options)
+    const resJson = await response.json()
+
+    return resJson
+}
+
+/*
+    Function to manually add Image to Firebase storage
+*/
+async function addImageManually (id, image, user, prompt ) {
+    const url = `${import.meta.env.VITE_BACKEND_URL}/generate/upload-firebase`
+
+    let options = {
+        method: "POST",
+        headers: {
+            'accept': 'application/json',
+            "Content-Type": "application/json",
+        },
+        body : JSON.stringify({
+            'imageUrl' : image,
+            'imageId' : id,
+            'imagePrompt' : prompt,
+            'userId' : user
+        })
+    }
+
+    const response = await fetch(url, options)
+    const resJson = await response.json()
+
+    return resJson
+}
+
+
+
+
+export default function  ArtImage(props) {
+    const [imageUrl, setImageUrl] = useState("")
+    const [allData, setAllData] = useState({})
+    const [generatedData, setGenData] = useState({})
+    const [selected, setSelected] = useState(false)
+
+    const imageData = props.imageData;
+
+    let waitTimeOut;
+    let countingTimeout;
+
+
+    // might keep this state to show remaining time
+    // TODO: replace with loading later
+    const [waitTime, setWaitTime] = useState(Infinity)
+
+    const cookies = new Cookies(null, { path : "/"})
+
+    /*
+        Recursive function to get firebase image url
+        Only called when image is still processing
+    */
+    async function setTimeoutFunction (wait) {
```

Comment:
For this function, I understand we setTimeout for the amount of time we expect to wait for, 

I think if we want a "live timer", we can use the waitTime you defined as input to the component that renders the time left. Then we can useEffect with `setInterval` for 1 second, every second, we decrement our `waitTime` state. When we reach 0, we can re-ping the `setTimeoutFunction`.  Does this help solve the timer issue?